purchases/src/main/kotlin/com/revenuecat/purchases/Offerings.kt

```diff
@@ -24,4 +25,59 @@ data class Offerings(
      * @param identifier Offering identifier
      */
     operator fun get(identifier: String) = getOffering(identifier)
+
+    /**
+     * Retrieves an specific offering by a placement identifier.
+     * For more info see https://www.revenuecat.com/docs/tools/targeting
+     * @param placementId Placement identifier
+     */
+    fun getCurrentOfferingForPlacement(placementId: String): Offering? {
+        val placements = this.placements ?: run {
+            return null
+        }
+
+        val offeringForPlacement = placements.offeringIdsByPlacement[placementId]?.let { getOffering(it) }
+        val fallbackOffering = placements.fallbackOfferingId?.let { getOffering(it) }
+
+        val showNoOffering = placements.offeringIdsByPlacement.containsKey(placementId)
```

Comment:
Nitpick but this only needs to be calculated if the offering by placement is null right? Should we do an if-else instead to avoid computing these when not needed?

Reply:
Yeah yeah, this could debatably go place the`if (showNoOffering)` line below! I pulled this out because I wanted to be explicit with the variable name on what this logic was doing 🤷‍♂️  The method will only get called explicitly be the developer and its a O(1) lookup so it shouldn't be anything too bad for unnecessary computing.

But I'm happy to move it in and add a comment if you'd think that would be better 😊

Reply:
NABD either way, just to avoid a couple extra ops, so I'm ok with this.